_distutils_hack/__init__.py

```diff
@@ -58,7 +58,9 @@ def ensure_local_distutils():
 
     # check that submodules load as expected
     core = importlib.import_module('distutils.core')
-    assert '_distutils' in core.__file__, core.__file__
+    # FIXME: this assertion blows up if the MetaFinder below has no-opped on a
+    #  setuptools from another path
+    # assert '_distutils' in core.__file__, core.__file__
```

Comment:
I'm trying to understand why this check is a problem. `ensure_local_distutils` is only called by `do_override`, which is only called by importing setuptools. But if an old version of setuptools is earlier on sys.path, it will not invoke the override.

Reply:
In 41f61f3cc, I've removed the commented code and restored the expectation to see what breaks.

Reply:
Aha. I see now (in the failed RTD build). Because the check for "matching setuptools" was very particular, and because another copy of Setuptools was taking precedence, _and_ because local checkouts don't get the `.pth` treatment, the shim wasn't getting loaded for any invocation (I think).

In da8c68c49, I'm exploring allowing any setuptools to satisfy the distutils requirement.